### PR TITLE
Split extensions into its own crates

### PR DESCRIPTION
**Pull request by hannesdejager**
This MR splits the GCS back-end and the Rest Authenticator into its own crates that simply depend on libunftp.

In the beginning phases of development it was convenient to put all of this code in the same crate. Now there is benefits to splitting it out. For example, if splitted then:

- libunftp can become stable 1.0 before the extensions are
- libunftp version don't need to get a major bump if one if the extension has a breaking change.
- the list of libunftp dependencies are smaller
- if we do it right every extension crate on crates.io creates visibility and shows examples to other people of how a libunftp extension crate would look like.
- More focus with less code in each crate.

### CODE REVIEW

File: crates/unftp-auth-rest/Makefile [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+	cargo fmt --all -- --check
+	cargo clippy --all-features --all-targets
+	cargo test --all-features --all-targets
+	cargo test --doc
+	cargo build --all-features --all-targets
+	cargo doc --all-features --no-deps
+
+.PHONY: publish
+publish: # Publishes the lib to crates.io
+	cargo publish --all-features --verbose
```
Comment (Nonius) [inline crates/unftp-auth-rest/Makefile:22]: question(non-blocking): what about removing the `--all-features` flag? it looks to me that this cargo file doesn't have any features, so this is unnecessary.

File: crates/unftp-sbe-gcs/Makefile [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+	cargo fmt --all -- --check
+	cargo clippy --all-features --all-targets
+	cargo test --all-features --all-targets
+	cargo test --doc
+	cargo build --all-features --all-targets
+	cargo doc --all-features --no-deps
+
+.PHONY: publish
+publish: # Publishes the lib to crates.io
+	cargo publish --all-features --verbose
```
Comment (Nonius) [inline crates/unftp-sbe-gcs/Makefile:22]: question(non-blocking): what about removing the --all-features flag? it looks to me that this cargo file doesn't have any features, so this is unnecessary.

File: crates/unftp-auth-rest/Makefile (hunk 1 above)
Comment (hannesdejager) [reply crates/unftp-auth-rest/Makefile:22]: yup, will remove

Comment (hannesdejager) [reply crates/unftp-auth-rest/Makefile]: Fixed

File: crates/unftp-sbe-gcs/Makefile (hunk 2 above)
Comment (hannesdejager) [reply crates/unftp-sbe-gcs/Makefile]: Fixed

**Pull request merged**